javagen/src/main/java/com/azure/autorest/mapper/CustomClientParameterMapper.java

```diff
@@ -28,11 +28,14 @@ public static CustomClientParameterMapper getInstance() {
 
     @Override
     public ClientMethodParameter map(Parameter parameter) {
+        return map(parameter, false);
+    }
+
+    public ClientMethodParameter map(Parameter parameter, boolean isProtocolMethod) {
```

Comment:
I added a `isProtocolMethod` boolean and the logic here.

@haolingdong-msft 
I am not sure about the background that we add this `CustomClientParameterMapper`. Is it purely for json-merge-patch? Can we merge it with `ClientParameterMapper`, or extract the shared logic (I assume at least half of the logic here is duplicated)?

Reply:
I can merge `CustomClientParameterMapper` with `ClientParameterMapper`, as there seems to be only one place of code diff.

But there is `CustomProxyParameterMapper` as well... Guess I would wait the background of these classes first.